### PR TITLE
Improve IronOS documentation

### PR DESCRIPTION
**Pull request by tr4nt0r**
## Proposed change

Adds sections with additional description, use cases, prerequisites, example automations, data updates, known limitations, troubleshooting and integration removal

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Added an "About IronOS" section detailing firmware features.
	- Introduced a "How you can use this integration" section for monitoring and controlling the soldering iron.
	- Added an "Automations" section with YAML configuration examples.

- **Documentation**
	- Included "Prerequisites" for Bluetooth range and ESPHome Bluetooth proxy.
	- Clarified the integration's discovery process for automatic detection of IronOS devices.
	- Added "Data updates" and "Known Limitations" sections.
	- Enhanced "Troubleshooting" section with solutions for common errors.

### CODE REVIEW

File: source/_integrations/iron_os.markdown [hunk 1]
```diff
@@ -20,10 +20,24 @@ ha_platforms:
...
+
+The IronOS integration lets you monitor and control your smart soldering iron from Home Assistant and automate related tasks. For example, you can turn on a fume extractor automatically when the soldering iron enters soldering mode, and turn it off when the iron is laid down. You can also monitor the temperature of the tip and handle, as well as power draw and input voltage, directly in dashboards.
+
 ## Minimum requirements
 
 - **Pinecil V2** requires IronOS v2.21 or higher to connect to Home Assistant. Please refer to the [Pinecil Wiki](https://wiki.pine64.org/wiki/Pinecil_Firmware#Overview) for update instructions.
 
+## Prerequisites
+
+The IronOS integration requires your device to be within Bluetooth range of Home Assistant, which must be equipped with a [Bluetooth adapter](https://www.home-assistant.io/integrations/bluetooth/). If the device is out of range, an [ESPHome Bluetooth proxy](https://esphome.io/projects/?type=bluetooth) can be placed within range. In this case, a Bluetooth adapter on Home Assistant is not required.
```
Comment (frenck) [inline source/_integrations/iron_os.markdown]:
```suggestion
The IronOS integration requires your device to be within Bluetooth range of Home Assistant, which must be equipped with a [Bluetooth adapter](/integrations/bluetooth/). If the device is out of range, an [ESPHome Bluetooth proxy](https://esphome.io/projects/?type=bluetooth) can be placed within range. In this case, a Bluetooth adapter on Home Assistant is not required.
```

File: source/_integrations/iron_os.markdown [hunk 2]
```diff
@@ -46,7 +60,93 @@ The **IronOS** {% term integration %} seamlessly connects Home Assistant with PI
...
+
+Automatically activate the fume extractor when soldering begins and deactivate it when the soldering iron is idle.
+
+{% my blueprint_import badge blueprint_url="https://community.home-assistant.io/t/ironos-soldering-fume-extractor-automation-pinecil-v2/802156" %}
+
+{% details "Example YAML configuration" %}
+
+{% raw %}
+
+```yaml
```
Comment (frenck) [inline source/_integrations/iron_os.markdown:81]: Please remove default values, and make sure it follows our YAML style guidelines: https://developers.home-assistant.io/docs/documenting/yaml-style-guide

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/iron_os.markdown [hunk 3]
```diff
@@ -46,7 +60,86 @@ The **IronOS** {% term integration %} seamlessly connects Home Assistant with PI
...
+  ```yaml
+  esp32:
+    board: ${board}
+    framework:
+      type: esp-idf
+      sdkconfig_options:
+        CONFIG_BT_GATTC_MAX_CACHE_CHAR: "100"
+  ```
+
+## Remove integration
```
Comment (klaasnicolaas) [inline source/_integrations/iron_os.markdown]:
```suggestion
## Remove the integration
```

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

**Review by frenck — APPROVED** [review]
Thanks, @tr4nt0r 👍 

../Frenck

**Review by frenck — CHANGES_REQUESTED** [review]
@tr4nt0r Contents looks good, but there is an merge conflict.

Can you take a look?

Thanks! 👍 

../Frenck

**Review by frenck — APPROVED** [review]
Thanks, @tr4nt0r 👍 

../Frenck

**Pull request merged**